prusti-viper/src/encoder/mir/procedures/encoder/lifetimes.rs

```diff
@@ -391,4 +448,219 @@ impl<'p, 'v: 'p, 'tcx: 'v> LifetimesEncoder for ProcedureEncoder<'p, 'v, 'tcx> {
             .filter(|x| !old_original_lifetimes.contains(x))
             .collect()
     }
+
+    fn encode_lifetime_specifications(
+        &mut self,
+    ) -> SpannedEncodingResult<(Vec<vir_high::Statement>, Vec<vir_high::Statement>)> {
+        let (first_bb, _) = rustc_middle::mir::traversal::reverse_postorder(self.mir)
+            .into_iter()
+            .next()
+            .unwrap();
+        let first_location = mir::Location {
+            block: first_bb,
+            statement_index: 0,
+        };
+
+        // Precondition: Inhale LifetimeTokens
+        let mut preconditions = vec![vir_high::Statement::comment(
+            "Lifetime preconditions.".to_string(),
+        )];
+        let lifetimes_to_inhale: BTreeSet<vir_high::ty::LifetimeConst> =
+            self.lifetimes_to_inhale()?;
+        for lifetime in &lifetimes_to_inhale {
+            let inhale_statement = self.encode_inhale_lifetime_token(lifetime.clone(), 1)?;
```

Comment:
```suggestion
            let inhale_statement = self.encode_inhale_lifetime_token(lifetime.clone(), rd)?;
```
This should be not `1`, but some unknown positive permission amount. You can use an unconstrained variable of type [`MPerm`](https://github.com/viperproject/prusti-dev/blob/15f651bc3cfffd78583b3862f2be9e0ad9b4feee/vir/defs/high/ast/ty.rs#L14-L15) for this.

Reply:
done